src/Stratis.Bitcoin.Features.BlockStore.Tests/BlockStoreTests.cs

```diff
@@ -0,0 +1,241 @@
+            this.repository.BlockHash = initializationHeader.HashBlock;
+
+            await this.blockStore.InitializeAsync().ConfigureAwait(false);
+            Assert.Equal(initializationHeader, this.blockStore.StoreTip);
+        }
+        
+        [Fact]
+        public async Task BlockStoreSavesBatchRightAwayWhenAtConsensusTipAsync()
+        {
+            this.repository.BlockHash = this.chain.Tip.Previous.HashBlock;
+
+            await this.blockStore.InitializeAsync().ConfigureAwait(false);
+
+            this.blockStore.AddToPending(new BlockPair(new Block(), this.chain.Tip));
+
+            await this.WaitUntilQueueIsEmpty().ConfigureAwait(false);
+            
+            Assert.Equal(this.blockStore.StoreTip, this.chain.Tip);
+            Assert.Equal(this.chain.Tip.HashBlock, this.repository.BlockHash);
+        }
+
+        [Fact]
+        public async Task BatchIsSavedAfter5MbLimitReachedAsync()
+        {
+            Block block = Block.Load(Encoders.Hex.DecodeData("000000202f6f6a130549473222411b5c6f54150d63b32aadf10e57f7d563cfc7010000001e28204471ef9ef11acd73543894a96a3044932b85e99889e731322a8ec28a9f9ae9fc56ffff011d0011b40202010000000001010000000000000000000000000000000000000000000000000000000000000000ffffffff2c028027266a24aa21a9ed09154465f26a2a4144739eba3e83b3e9ae6a1f69566eae7dc3747d48f1183779010effffffff0250b5062a0100000023210263ed47e995cbbf1bc560101e3b76c6bdb1b094a185450cea533781ce598ff2b6ac0000000000000000266a24aa21a9ed09154465f26a2a4144739eba3e83b3e9ae6a1f69566eae7dc3747d48f1183779012000000000000000000000000000000000000000000000000000000000000000000000000001000000000101cecd90cd38ac6858c47f2fe9f28145d6e18f9c5abc7ef1a41e2f19e6fe0362580100000000ffffffff0130b48d06000000001976a91405481b7f1d90c5a167a15b00e8af76eb6984ea5988ac0247304402206104c335e4adbb920184957f9f710b09de17d015329fde6807b9d321fd2142db02200b24ad996b4aa4ff103000348b5ad690abfd9fddae546af9e568394ed4a83113012103a65786c1a48d4167aca08cf6eb8eed081e13f45c02dc6000fd8f3bb16242579a00000000"), Network.Main);
+            int blockSize = block.GetSerializedSize();
+
+            int count = (5 * 1000 * 1000) / blockSize + 2;
+
+            ConcurrentChain longChain = this.CreateChain(count);
+            this.consensusTip = longChain.Tip;
+            this.repository.BlockHash = longChain.Genesis.HashBlock;
+
+            this.blockStore = new BlockStore(this.repository, longChain, this.chainState, new StoreSettings(), new NodeLifetime(), new LoggerFactory());
+
+            await this.blockStore.InitializeAsync().ConfigureAwait(false);
+
+            for (int i = 1; i < count; i++)
+            {
+                ChainedHeader header = longChain.GetBlock(i);
+
+                this.blockStore.AddToPending(new BlockPair(block, header));
+            }
+            
+            await this.WaitUntilQueueIsEmpty().ConfigureAwait(false);
+            Assert.Equal(this.blockStore.StoreTip, longChain.GetBlock(count - 1));
```

Comment:
Is this assert equal the right way round?  I can't quite work out which one is being asserted. Also, why the -1  on the GetBlock(count -1). Could this be made clearer?

Reply:
>Also, why the -1 on the GetBlock(count -1).

Because we didn't push the block that represent's the consensus tip. 
When consensus tip is pushed the batch size is always ignored and saved right away

>Is this assert equal the right way round? 

Nope. Fixed